src/sqlfluff/core/parser/lexer.py

```diff
@@ -118,6 +118,9 @@ def __bool__(self) -> bool:
         return len(self.elements) > 0
 
 
+LexerType = Union["RegexLexer", "StringLexer"]
```

Comment:
Is this necessary? `StringLexer` inherits from `RegexLexer` so I would have thought that just saying `StringLexer` is enough for type inheritance.

It could be clearer though I suppose if we had a `BaseLexer` which both inherited from, but that might add more complexity than we need.